ksim/observation.py

```diff
@@ -256,6 +257,42 @@ def observe(self, state: ObservationState, rng: PRNGKeyArray) -> Array:
             return state.physics_state.data.qacc
 
 
+@attrs.define(frozen=True, kw_only=True)
+class ContactObservation(Observation):
+    geom_idxs: tuple[int, ...] = attrs.field()
+    custom_name: str | None = attrs.field(default=None)
```

Comment:
`custom_name` makes me so sad :( the user shouldn't be providing custom names in my opinion - if there are duplicates of the same class, then we should override the `get_name` function to use some information about the object to disambiguate the name, but we should not make the user provide the full name